tests/PHPStan/Analyser/data/generic-callables.php

```diff
@@ -78,3 +78,146 @@ function testNestedClosures(Closure $closure, string $str, int $int): void
 	$result = $closure1($int);
 	assertType('int|string', $result);
 }
+
+/**
+ * @template T
+ * @param T $arg
+ * @return T
+ */
+function foo(mixed $arg): mixed {}
+
+class Foo
+{
+	/**
+	 * @template T
+	 * @param T $arg
+	 * @return T
+	 */
+	public function foo(mixed $arg): mixed {}
+}
+
+function test(): void
+{
+	assertType('Closure<T of mixed>(T): T', foo(...));
+	assertType('1', foo(...)(1));
+
+	$foo = new Foo();
+	$closure = Closure::fromCallable([$foo, 'foo']);
+	assertType('Closure<T of mixed>(T): T', $closure);
+	assertType('1', $closure(1));
+}
+
+/**
+ * @template A
+ * @param A $value
+ * @return A
+ */
+function identity(mixed $value): mixed
+{
+	return $value;
+}
+
+/**
+ * @template B
+ * @param B $value
+ * @return B
+ */
+function identity2(mixed $value): mixed
+{
+	return $value;
+}
+
+function testIdentity(): void
+{
+	assertType('array{1, 2, 3}', array_map(identity(...), [1, 2, 3]));
+}
+
+/**
+ * @template A
+ * @template B
+ * @param A $value
+ * @param B $value2
+ * @return A|B
+ */
+function identityTwoArgs(mixed $value, mixed $value2): mixed
+{
+	return $value || $value2;
+}
+
+function testIdentityTwoArgs(): void
+{
+	assertType('non-empty-array<int, 1|2|3|4|5|6>', array_map(identityTwoArgs(...), [1, 2, 3], [4, 5, 6]));
+}
+
+/**
+ * @template A
+ * @template B
+ * @param list<A> $a
+ * @param list<B> $b
+ * @return list<array{A, B}>
+ */
+function zip(array $a, array $b): array
+{
+}
+
+function testZip(): void
+{
+	$fn = zip(...);
+
+	assertType('list<array{1, 2}>', $fn([1], [2]));
+}
+
+/**
+ * @template X
+ * @template Y
+ * @template Z
+ * @param callable(X $x, Y $y): Z $fn
+ * @return callable(Y $xy, X $yx): Z
```

Comment:
without parameter names on the callable then the generic type resolution fails, because `GenericParametersAcceptorResolver` uses `$nativeParameterReflection->getName()` which returns `''` for unnamed parameters, and uses those names as array keys. If multiple parameters are unnamed this results in assigning to `$namedArgTypes['']` repeatedly and it falls apart

I've tried taking a stab at making `GenericParametersAcceptorResolver` care less about parameter names, but it's getting the better of me. Still trying to have a go at solving it but thought i'd put this up to possibly get some wisdom

Reply:
scratch this, in rubber duck debugging fashion as i look at the code to describe the issue i had a wave of inspiration...